api/app/enums/ingestion_target_kind.rb

```diff
@@ -0,0 +1,77 @@
+class IngestionTargetKind < ClassyEnum::Base
+
+  def reingest?(ingestion)
+    ingestion.text.present?
+  end
+
+  # rubocop:disable Metrics/MethodLength
+  def begin_processing(user, ingestion)
+    serialization_options = { current_user: user }
+    serialization = V1::IngestionSerializer.new(
+      ingestion,
+      serialization_options
+    ).serializable_hash
+    IngestionChannel.broadcast_to ingestion, type: "entity", payload: serialization
+
+    outcome = nil
+
+    begin
+      PaperTrail.request(whodunnit: user) do
+        outcome = Ingestions::Ingestor.run ingestion: ingestion
+      end
+      outcome
+    rescue StandardError => e
+      return ingestion.handle_ingestion_exception(e)
+    end
+
+    if outcome.valid?
+      ingestion.info("\nIngestion Complete.")
+      ingestion.text = outcome.result
+      ingestion.processing_success
+    else
+      ingestion.handle_ingestion_exception(outcome.errors)
+    end
+  end
+  # rubocop:enable Metrics/MethodLength
+end
+
+class IngestionTargetKind::Text < IngestionTargetKind
+end
+
+class IngestionTargetKind::TextSection < IngestionTargetKind
+  def reingest?(ingestion)
+    ingestion.text_section.present?
+  end
+
+  # rubocop:disable Metrics/MethodLength
+  def begin_processing(user, ingestion)
```

Comment:
Is there any difference between the two `begin_processing` methods in these classes other than the fact that one sets `ingestion.text` and the other sets `ingestion.text_section`?

If not, I would remove the redundant code in here by doing something like this:

```
class IngestionTargetKind < ClassyEnum::Base
  [...]
  # rubocop:disable Metrics/MethodLength
  def begin_processing(user, ingestion)
    serialization_options = { current_user: user }
    serialization = V1::IngestionSerializer.new(
      ingestion,
      serialization_options
    ).serializable_hash
    IngestionChannel.broadcast_to ingestion, type: "entity", payload: serialization

    outcome = nil

    begin
      PaperTrail.request(whodunnit: user) do
        outcome = Ingestions::Ingestor.run ingestion: ingestion
      end
      outcome
    rescue StandardError => e
      return ingestion.handle_ingestion_exception(e)
    end

    if outcome.valid?
      ingestion.info("\nIngestion Complete.")
      attach_result(ingestion, outcome.result)
      ingestion.processing_success
    else
      ingestion.handle_ingestion_exception(outcome.errors)
    end
  end
  # rubocop:enable Metrics/MethodLength

  def attach_result(ingestion, result)
    raise NoMethodError("IngestionTargetKind enum must implement attach_result")
  end

  [...]

end

class IngestionTargetKind::Text < IngestionTargetKind

  def attach_result(ingestion, result)
    ingestion.text = result
  end

end

class IngestionTargetKind::TextSection < IngestionTargetKind

  def attach_result(ingestion, result)
    ingestion.text_section = result
  end

end

Reply:
(I might also use this as an opportunity to break this long method into multiple methods to avoid having to disable the MethodLength lint rule.)